tools/data/activitynet/convert_proposal_format.py

```diff
@@ -0,0 +1,132 @@
+    """Read predictions from result dict."""
+    proposals = {}
+    num_proposals = 0
+    for video_id in result_dict:
+        result = result_dict[video_id]
+        this_video_proposals = []
+        for proposal in result:
+            t_start, t_end = proposal['segment']
+            score = proposal['score']
+            this_video_proposals.append([t_start, t_end, score])
+            num_proposals += 1
+        proposals[video_id] = np.array(this_video_proposals)
+    return proposals, num_proposals
+
+
+def dump_classifier_input(video_idx, video_id, num_frames, fps, gts, proposals,
+                          tiou, t_overlap_self, action_classifier_input):
+    """dump the input proposal file of action classifier (e.g: SSN).
+
+    Args:
+        video_idx (int): Index of video.
+        video_id (str): ID of video.
+        num_frames (int): Total frames of the video.
+        fps (float): Fps of the video.
+        gts (np.ndarray[float]): t_start, t_end and label of groundtruths.
+        proposals (np.ndarray[float]): t_start, t_end and score of proposals.
+        tiou (np.ndarray[float]): 2-dim array with IoU ratio.
+        t_overlap_self (np.ndarray[float]): 2-dim array with overlap_self
+            (union / self_len) ratio.
+        action_classifier_input (open file object): Open file object of
+            action_classifier_input file.
+    """
+
+    action_classifier_input.write(
+        f'#{video_idx}\n{video_id}\n{num_frames}\n{fps}\n{gts.shape[0]}\n')
+    for gt in gts:
+        action_classifier_input.write(f'{int(gt[2])} {gt[0]} {gt[1]}\n')
+    action_classifier_input.write(f'{proposals.shape[0]}\n')
+
+    best_iou = np.amax(tiou, axis=0)
+    best_iou_index = np.argmax(tiou, axis=0)
+    best_overlap = np.amax(t_overlap_self, axis=0)
+    best_overlap_index = np.argmax(t_overlap_self, axis=0)
+
+    for i in range(proposals.shape[0]):
+        index_iou = best_iou_index[i]
+        index_overlap = best_overlap_index[i]
+        label_iou = gts[index_iou][2]
+        label_overlap = gts[index_overlap][2]
+        if label_iou != label_overlap:
+            label = label_iou if label_iou != 0 else label_overlap
+        else:
+            label = label_iou
+        if best_iou[i] == 0 and best_overlap[i] == 0:
+            action_classifier_input.write(
+                f'0 0 0 {proposals[i][0]} {proposals[i][1]}\n')
+        else:
+            action_classifier_input.write(
+                f'{int(label)} {best_iou[i]} {best_overlap[i]} '
+                f'{proposals[i][0]} {proposals[i][1]}\n')
+
+
+data_path = '../../../data/ActivityNet'
+result_path = '../../..'
+proposal_generator_output_file = f'{result_path}/results.json'
+action_classifier_input_file = f'{result_path}/anet_val_classifier_input.txt'
+activity_index_file = f'{data_path}/anet_activity_indexes_val.txt'
+ann_file = f'{data_path}/anet_anno_val.json'
+
+action_classifier_input = open(action_classifier_input_file, 'w')
```

Comment:
Still, do not call it `xxx_input`, it is the input of another module, but the output of this file. Called it `formatted_proposal_file`